docs/index.rst

```diff
@@ -18,6 +18,7 @@ Contents:
    list_of_services
    common_tasks
    rest_api
+   submitting_pulse_data
```

Comment:
Could we just make them two subsections under "Submitting Data": 

* Submitting using the Python client
* Submitting using Pulse

Then you could have an intro section under "submitting data" describing the advantages/disadvantages of each.

IMO it's very confusing to put this information in two different sections. Someone looking at the Treeherder documentation for the first time will very likely have no idea what "pulse" is.